config/CLIConfiguration.ts

```diff
@@ -228,23 +237,66 @@ class _CLIConfiguration {
   }
 
   getDefaultAccount(): string | number | null {
-    return this.config && this.config.defaultAccount
-      ? this.config.defaultAccount
-      : null;
+    return this.getCWDAccountOverride() || this.config?.defaultAccount || null;
+  }
+
+  getDefaultAccountOverrideFilePath(): string | null {
+    return findup([DEFAULT_ACCOUNT_OVERRIDE_FILE_NAME], {
+      cwd: getCwd(),
+    });
   }
 
-  // TODO a util that returns the account to use, respecting the values set in
-  // "defaultAccountOverrides"
-  // Example "defaultAccountOverrides":
-  //  - /src/brodgers/customer-project-1: customer-account1
-  //  - /src/brodgers/customer-project-2: customer-account2
-  // "/src/brodgers/customer-project-1" is the path to the project dir
-  // "customer-account1" is the name of the account to use as the default for the specified dir
-  // These defaults take precedence over the standard default account specified in the config
-  getResolvedDefaultAccountForCWD(
-    nameOrId: string | number
-  ): CLIAccount_NEW | null {
-    return this.getAccount(nameOrId);
+  getCWDAccountOverride(): string | number | null {
+    const defaultOverrideFile = this.getDefaultAccountOverrideFilePath();
+    if (!defaultOverrideFile) {
+      return null;
+    }
+    try {
+      const source = fs.readFileSync(defaultOverrideFile, 'utf8');
+      const accountId = Number(source);
+
+      if (isNaN(accountId)) {
+        throw new Error(
+          i18n(`${i18nKey}.getCWDAccountOverride.errorHeader`, {
+            hsAccountFile: defaultOverrideFile,
+          }),
+          {
+            cause: DEFAULT_ACCOUNT_OVERRIDE_ERROR_INVALID_ID,
+          }
+        );
+      }
```

Comment:
Throwing inside in a `try` block is an anti-pattern because it essentially acts as a `goto` to the corresponding `catch` block.  It seems like we could just wrap this line in the `try` block
```
const source = fs.readFileSync(defaultOverrideFile, 'utf8');
```

and move the these two if statements outside of the `try` block entirely and throw the errors to be handled by the invoking method.